server/src/main/kotlin/org/itmodreamteam/myrest/server/service/restaurant/RestaurantServiceImpl.kt

```diff
@@ -0,0 +1,122 @@
+package org.itmodreamteam.myrest.server.service.restaurant
+
+import org.itmodreamteam.myrest.server.error.UserException
+import org.itmodreamteam.myrest.server.model.restaurant.Restaurant
+import org.itmodreamteam.myrest.server.repository.restaurant.RestaurantRepository
+import org.itmodreamteam.myrest.shared.restaurant.RestaurantInfo
+import org.itmodreamteam.myrest.shared.restaurant.RestaurantRegistrationInfo
+import org.itmodreamteam.myrest.shared.restaurant.RestaurantStatus
+import org.itmodreamteam.myrest.shared.restaurant.RestaurantUpdateInfo
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.Pageable
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import javax.validation.ConstraintViolationException
+
+@Service
+class RestaurantServiceImpl (
+    private val restaurantRepository: RestaurantRepository,
+) : RestaurantService {
+
+    override fun register(newRestaurant: RestaurantRegistrationInfo): RestaurantInfo {
+        val existingRestaurant = restaurantRepository.findByName(newRestaurant.name)
+        if (existingRestaurant == null) {
+            try {
+                val restaurant = restaurantRepository.save(Restaurant(newRestaurant))
+                return RestaurantInfo(
+                    restaurant.id,
+                    restaurant.status,
+                    restaurant.name,
+                    restaurant.description,
+                    restaurant.legalInfo,
+                    restaurant.websiteUrl,
+                    restaurant.phone,
+                    restaurant.email,
+                    restaurant.internalRating,
+                    restaurant.externalRating
+                )
+            } catch (e: ConstraintViolationException) {
+                throw UserException("Неверное значение поля")
```

Comment:
No point in wrapping exceptions at the service layer, we'll do this at the REST controllers layer